### PR TITLE
Add Module 3 tasks

### PR DESCRIPTION
**Pull request by MiraMakh**

### CODE REVIEW

File: src/components/MovieTitle/MovieTitle.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      <div className={styles.movieTile__info}>
+        <h3 className={styles.movieTile__title}>{title}</h3>
+        <p className={styles.movieTile__yearGenres}>
+          {releaseYear} ・ {genres.join(', ')}
+        </p>
+      </div>
+      <button
+        className={styles.movieTile__menuButton}
+        onClick={toggleContextMenu}
+      >
```
Comment (femiarkh) [inline src/components/MovieTitle/MovieTitle.tsx]: It is a good idea to always provide a type for a button element. Otherwise, by default it will be considered as of "submit" type, which is probably not what you want here.
You can read more [here](https://dev.to/clairecodes/why-its-important-to-give-your-html-button-a-type-58k9).

File: src/components/MovieTitle/MovieTitle.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  const handleEditClick = () => {
+    if (onEdit) onEdit(id);
+    setIsContextMenuOpen(false);
+  };
+
+  const handleDeleteClick = () => {
+    if (onDelete) onDelete(id);
+    setIsContextMenuOpen(false);
+  };
```
Comment (femiarkh) [inline src/components/MovieTitle/MovieTitle.tsx]: You are stopping propagation when clicking the Menu button, but how about the Edit and Delete buttons? Do you want clicking them to also trigger `onClick` callback?

File: src/components/MovieTitle/MovieTitle.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  return (
+    <div
+      className={clsx(styles.movieTile, {
+        [styles.movieTile__active]: isActive,
```
Comment (femiarkh) [inline src/components/MovieTitle/MovieTitle.tsx]: There is no such style in `MovieTile.module.scss`, so you're getting `undefined` as a class name. That's why it is a good idea to configure type-safe CSS modules.

File: src/components/SortControl/SortControl.module.scss [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+.sortControl__select {
+  padding: toRem(5) toRem(10);
+  font-size: 14px;
+  border: 1px solid #ccc;
+  border-radius: 5px;
+  background: white;
+  color: #333;
+
+  &:focus {
+    outline: none;
```
Comment (femiarkh) [inline src/components/SortControl/SortControl.module.scss]: You shouldn't hide outline completely, this is not good from a11y perspective.

File: src/components/SortControl/SortControl.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+    expect(screen.getByLabelText('Sort by')).toBeInTheDocument();
+
+    expect(screen.getByRole('combobox')).toBeInTheDocument();
+    expect(
+      screen.getByRole('option', { name: 'Release Date' })
+    ).toBeInTheDocument();
+    expect(screen.getByRole('option', { name: 'Title' })).toBeInTheDocument();
+  });
+});
```
Comment (femiarkh) [inline src/components/SortControl/SortControl.spec.tsx:24]: I would also check that the dropdown behaves correctly, i.d. calls `onChange` with a chosen value.

File: src/components/MovieTitle/MovieTile.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+    /* @TODO: fix this test fails */
+    const menuButton = screen.getByText('...');
+    await userEvent.click(menuButton);
+
+    const editButton = screen.getByText('Edit');
+    await userEvent.click(editButton);
+    expect(mockOnEdit).toHaveBeenCalledWith(movie.id);
+
+    const deleteButton = screen.getByText('Delete');
```
Comment (femiarkh) [inline src/components/MovieTitle/MovieTile.spec.tsx]: The menu is closed on Edit click, so there is no Delete button in the DOM and the test fails.

File: src/components/MovieDetails/MovieDetails.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import { render, screen } from '@testing-library/react';
+import MovieDetails from './MovieDetails';
+import { MovieDetailsProps } from '../../models';
+
+describe('MovieDetails Component', () => {
+  const defaultProps: MovieDetailsProps = {
+    imageUrl: `./src/assets/Bitmap.png`,
+    title: 'Movie title',
+    rating: 8.8,
+    duration: '2h 34min',
```
Comment (femiarkh) [inline src/components/MovieDetails/MovieDetails.spec.tsx:10]: The value is going to come as a number from BE, so be prepared to convert it to a string by yourself (it is not a problem if you do it later).

File: src/components/MovieTitle/MovieTitle.tsx [hunk 8]
```diff
@@ -0,0 +1,64 @@
+import React, { useState } from 'react';
```
Comment (femiarkh) [inline src/components/MovieTitle/MovieTitle.tsx]: I guess the name of the folder and the file should be `Movie Tile`, not `Movie Title`.

**Review by femiarkh — COMMENTED** [review]
Good job!
Please fix the failing unit test.
You could also add a couple of additional tests checking edge cases, but this is optional.

File: src/components/MovieTitle/MovieTitle.tsx (hunk 2 above)
Comment (MiraMakh) [reply src/components/MovieTitle/MovieTitle.tsx]: I thought to close menu as I am redirected on edit functionality and no need of this menu

File: src/components/MovieTitle/MovieTitle.tsx (hunk 3 above)
Comment (MiraMakh) [reply src/components/MovieTitle/MovieTitle.tsx]: I was going to delate at this point and missed it, will remove active class

File: src/components/SortControl/SortControl.spec.tsx (hunk 5 above)
Comment (MiraMakh) [reply src/components/SortControl/SortControl.spec.tsx:24]: I will add this a little late will left TODO comment for it

File: src/components/MovieTitle/MovieTile.spec.tsx (hunk 6 above)
Comment (MiraMakh) [reply src/components/MovieTitle/MovieTile.spec.tsx]: Thank you good catch

File: src/components/MovieTile/MovieTile.tsx [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        type="button"
+        className={styles.movieTile__menuButton}
+        onClick={toggleContextMenu}
+      >
+        ...
+      </button>
+      {isContextMenuOpen && (
+        <div className={styles.movieTile__contextMenu}>
+          <button onClick={handleEditClick}>Edit</button>
+          <button onClick={handleDeleteClick}>Delete</button>
```
Comment (femiarkh) [inline src/components/MovieTile/MovieTile.tsx:52]: Same for these buttons - please provide `type` attribute.

File: src/components/MovieTitle/MovieTitle.tsx (hunk 2 above)
Comment (femiarkh) [reply src/components/MovieTitle/MovieTitle.tsx]: Yes, closing the menu is the right thing to do. I'm talking about triggering `onClick` callback by propagating the click to the card wrapper in addition to executing `handleEditClick` or `handleDeleteClick` callbacks. So, for example, if clicking the card opens the movie's Details page, then it will open whenever the editing modal or deleting dialog is open. If it is your intention, then it's ok, but if not, you should also stop propagation in these two callbacks.

**Review by femiarkh — APPROVED** [review]

File: src/components/MovieTile/MovieTile.tsx (hunk 9 above)
Comment (MiraMakh) [reply src/components/MovieTile/MovieTile.tsx:52]: Will do in new PR

File: src/components/MovieTitle/MovieTitle.tsx (hunk 2 above)
Comment (MiraMakh) [reply src/components/MovieTitle/MovieTitle.tsx]: Will do in new PR
